executor/src/witgen/affine_expression.rs

```diff
@@ -19,6 +20,21 @@ pub enum AffineExpression<K, T> {
     ManyVars(Vec<(K, T)>, T),
 }
 
+#[derive(Debug, Clone, PartialEq, PartialOrd, Eq, Ord, Copy)]
+pub enum AlgebraicVariable<'a> {
+    Reference(&'a AlgebraicReference),
+    Public(&'a str),
```

Comment:
They do have IDs as well...

Reply:
Hmm, the reason it's `&str` is because `AlgebraicExpression::PublicReference` stores a string. So I think this should be changed either before or after this PR.

Reply:
I mean before this PR it did not matter for performance.

Reply:
I think the change is not too big: #1759

When this is merged, I can rebase and change it to `u64` here too.